src/engine/srv_internal.h

```diff
@@ -173,6 +174,19 @@ int sched_req_enqueue(struct dss_xstream *dx, struct sched_req_attr *attr,
 		      void (*func)(void *), void *arg);
 void sched_stop(struct dss_xstream *dx);
 
+
+static inline bool
+sched_xstream_stopping(void)
+{
+	struct dss_xstream	*dx = dss_current_xstream();
+	ABT_bool		 state;
+	int			 rc;
+
+	rc = ABT_future_test(dx->dx_stopping, &state);
```

Comment:
I do not see where do you wait for this future? then why do you use future here?

Reply:
Nobody wait for this, it's just a future test, similar to dx_shutdown. What do you suggest rather than future?